src/parser-parse5.js

```diff
@@ -6,7 +6,8 @@ function parse(text) {
   // Inline the require to avoid loading all the JS if we don't use it
   const parse5 = require("parse5");
   try {
-    const ast = parse5.parse(text, {
+    const isFragment = !/^\s*<(!doctype|html|head|body)/i.test(text);
```

Comment:
I would have a failback where if it throws, try the other one. This way the regex doesn't have to be perfect.